### PR TITLE
feat: Adding UI for NL2F Expression Assistant and Copilot Service Interface

### PR DESCRIPTION
**Pull request by maturpi**
- **Please check if the PR fulfills these requirements**

* [ ] The commit message follows our guidelines
* [ ] Tests for the changes have been added (for bug fixes/features)
* [ ] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

- **What is the current behavior?** (You can also link to an open issue here)
N/A

- **What is the new behavior (if this is a feature change)?**
N/A

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

- **Please Include Screenshots or Videos of the intended change**:
nl2fExpressionUIDemo

### CODE REVIEW

**Comment by hartra344** [comment]
please add some tests

**Review by LLotme — COMMENTED** [review]
gh pr checkout 5107

**Comment by maturpi** [comment]
> please add some tests

Done @hartra344

**Comment by hartra344** [comment]
@Eric-B-Wu can you review please?

File: libs/logic-apps-shared/src/designer-client-services/lib/base/copilot.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export class BaseCopilotService implements ICopilotService {
+  constructor(public readonly options: CopilotServiceOptions) {}
+
+  async getNl2fExpressions(query: string, originalExpression?: string): Promise<Nl2fExpressionResult> {
+    const { isDev } = this.options;
+    if (isDev) {
+      return generateMockResults(query, originalExpression);
+    }
+
+    // To be implemented by LAUX team
```
Comment (Eric-B-Wu) [inline libs/logic-apps-shared/src/designer-client-services/lib/base/copilot.ts:16]: can we create a github issue/work item for this

**Review by Eric-B-Wu — APPROVED** [review]

**Pull request merged**